lib/grammers-client/src/client/channel.rs

```diff
@@ -0,0 +1,75 @@
+use super::Client;
+use anyhow::{anyhow, Result};
+use grammers_tl_types::{
+    enums::{
+        channels::ChannelParticipants::{NotModified, Participants},
+        messages::ChatFull,
+        ChannelParticipantsFilter,
+        InputChannel::Channel,
+        User,
+    },
+    functions::channels::{GetFullChannel, GetParticipants},
+    types::InputChannel,
+};
+
+impl Client {
+    const MAX_PARTICIPANT_LIMIT: usize = 200;
+
+    pub async fn get_full_channel(&mut self, channel_id: i64) -> Result<ChatFull> {
+        let input_channel = InputChannel {
+            channel_id,
+            access_hash: 0i64,
```

Comment:
This is unlikely to be reliable. The API generally expects you to know the access hash.